includes/class-frontend.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * Frontend functionality.
+ *
+ * @package AdvancedAds\Genesis
+ * @author  Advanced Ads <info@wpadvancedads.com>
+ * @since   1.0.0
+ */
+
+namespace AdvancedAds\Genesis;
+
+use AdvancedAds\Framework\Interfaces\WordPress_Integration;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Frontend Class.
+ */
+class Frontend implements WordPress_Integration {
+
+	/**
+	 * Hold genesis hooks from placements.
+	 *
+	 * @var array
+	 */
+	private $genesis_hooks = null;
+
+	/**
+	 * Hook into WordPress.
+	 *
+	 * @return void
+	 */
+	public function hooks() {
+		$hooks = array_keys( $this->get_genesis_hooks() );
+		foreach ( $hooks as $hook ) {
+			add_action( $hook, [ $this, 'execute_hook' ] );
+		}
+	}
+
+	/**
+	 * Execute genesis hooks
+	 *
+	 * @since 1.0.0
+	 *
+	 * @return void
+	 */
+	public function execute_hook() {
+		$current_hook = current_filter();
+		$hooks        = $this->get_genesis_hooks();
+
+		if ( in_array( $current_hook, $hooks, true ) ) {
+			the_ad_placement( $hooks[ $current_hook ] );
+		}
+	}
+
+	/**
+	 * Get genesis hooks from placements.
+	 *
+	 * @return array
+	 */
+	private function get_genesis_hooks() {
+		// Early bail!!
+		if ( null !== $this->genesis_hooks ) {
+			return $this->genesis_hooks;
+		}
```

Comment:
Can't this be an array from the beginning? And we check for emptiness instead of overwriting the type?